Signal/src/util/OWSBackupExport.m

```diff
@@ -0,0 +1,601 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+#import "OWSBackupExport.h"
+#import "Signal-Swift.h"
+#import "zlib.h"
+#import <Curve25519Kit/Randomness.h>
+#import <SSZipArchive/SSZipArchive.h>
+#import <SignalServiceKit/NSDate+OWS.h>
+#import <SignalServiceKit/OWSBackgroundTask.h>
+#import <SignalServiceKit/OWSBackupStorage.h>
+#import <SignalServiceKit/OWSError.h>
+#import <SignalServiceKit/OWSFileSystem.h>
+#import <SignalServiceKit/TSAttachmentStream.h>
+#import <SignalServiceKit/TSMessage.h>
+#import <SignalServiceKit/TSThread.h>
+#import <SignalServiceKit/Threading.h>
+#import <SignalServiceKit/YapDatabaseConnection+OWS.h>
+#import <YapDatabase/YapDatabase.h>
+#import <YapDatabase/YapDatabaseCryptoUtils.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef void (^OWSBackupExportBoolCompletion)(BOOL success);
+typedef void (^OWSBackupExportCompletion)(NSError *_Nullable error);
+
+#pragma mark -
+
+@interface OWSAttachmentExport : NSObject
```

Comment:
This class is used to lazily encrypt a copy of an attachment to disk, if necessary.